src/kaleidoscope/hardware/keyboardio/Model01.cpp

```diff
@@ -60,78 +45,51 @@ void Model01::enableHighPowerLeds(void) {
   // Set B4, the overcurrent check to an input with an internal pull-up
   DDRB &= ~_BV(4);	// set bit, input
   PORTB &= ~_BV(4);	// set bit, enable pull-up resistor
-
-
-
 }
 
-void Model01::setup(void) {
-  wdt_disable();
-  delay(100);
-  enableScannerPower();
-
+void Model01LEDDriver::setup() {
   // TODO: Consider not doing this until 30s after keyboard
   // boot up, to make it easier to rescue things in case of power draw issues.
   enableHighPowerLeds();
-  leftHandState.all = 0;
-  rightHandState.all = 0;
-
-  TWBR = 12; // This is 400mhz, which is the fastest we can drive the ATTiny
 }
 
-void Model01::enableHardwareTestMode() {
-
-  // Toggle the programming LEDS on
-  PORTD |= (1 << 5);
-  PORTB |= (1 << 0);
-
-  // Disable the debouncer on the ATTinys
-  KeyboardHardware.setKeyscanInterval(2);
-}
-
-
-
-void Model01::setCrgbAt(int8_t i, cRGB crgb) {
+void Model01LEDDriver::setCrgbAt(int8_t i, cRGB crgb) {
   if (i < 0) {
     return;
   }
   if (i < 32) {
     cRGB oldColor = getCrgbAt(i);
     isLEDChanged |= !(oldColor.r == crgb.r && oldColor.g == crgb.g && oldColor.b == crgb.b);
 
-    leftHand.ledData.leds[i] = crgb;
+    Model01KeyScanner::leftHand.ledData.leds[i] = crgb;
```

Comment:
Since the Model 01 is kind of a special snowflake, offloading both keyscanner and led driver to the same "other" chip, I know the naming doesn't quite work right. it's not Important, but we may want to consider how to make it clearer.

Reply:
One of the ideas I was toying with was having a `Model01Controller` class or something, to which both the LED driver and the KeyScanner can dispatch to. That's a bit more code and boilerplate, but would perhaps be clearer than the status quo.